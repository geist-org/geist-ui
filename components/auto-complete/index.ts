import AutoComplete from './auto-complete'
import AutoCompleteItem from './auto-complete-item'
import AutoCompleteSearching from './auto-complete-searching'
import AutoCompleteEmpty from './auto-complete-empty'

export type AutoCompleteComponentType = typeof AutoComplete & {
  Item: typeof AutoCompleteItem
  Option: typeof AutoCompleteItem
  Searching: typeof AutoCompleteSearching
  Empty: typeof AutoCompleteEmpty
}
;(AutoComplete as AutoCompleteComponentType).Item = AutoCompleteItem
;(AutoComplete as AutoCompleteComponentType).Option = AutoCompleteItem
;(AutoComplete as AutoCompleteComponentType).Searching = AutoCompleteSearching
;(AutoComplete as AutoCompleteComponentType).Empty = AutoCompleteEmpty

export type {
  AutoCompleteOption,
  AutoCompleteOptions,
  AutoCompleteProps,
  AutoCompleteTypes,
} from './auto-complete'
export default AutoComplete as AutoCompleteComponentType
