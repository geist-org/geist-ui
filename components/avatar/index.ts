import Avatar from './avatar'
import AvatarGroup from './avatar-group'

export type AvatarComponentType = typeof Avatar & {
  Group: typeof AvatarGroup
}
;(Avatar as AvatarComponentType).Group = AvatarGroup

export type { AvatarProps } from './avatar'
export type { AvatarGroupProps } from './avatar-group'

export default Avatar as AvatarComponentType
