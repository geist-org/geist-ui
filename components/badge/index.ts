import Badge from './badge'
import BadgeAnchor from './badge-anchor'

export type BadgeComponentType = typeof Badge & {
  Anchor: typeof BadgeAnchor
}
;(Badge as BadgeComponentType).Anchor = BadgeAnchor

export type { BadgeProps, BadgeTypes } from './badge'
export type { BadgeAnchorProps, BadgeAnchorPlacement } from './badge-anchor'
export default Badge as BadgeComponentType
