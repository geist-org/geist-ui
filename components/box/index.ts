import Box from './box'

export type { BoxProps } from './box'
export default Box
