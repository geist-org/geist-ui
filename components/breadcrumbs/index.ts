import Breadcrumbs from './breadcrumbs'
import BreadcrumbsItem from './breadcrumbs-item'
import BreadcrumbsSeparator from './breadcrumbs-separator'

export type BreadcrumbsComponentType = typeof Breadcrumbs & {
  Item: typeof BreadcrumbsItem
  Separator: typeof BreadcrumbsSeparator
}
;(Breadcrumbs as BreadcrumbsComponentType).Item = BreadcrumbsItem
;(Breadcrumbs as BreadcrumbsComponentType).Separator = BreadcrumbsSeparator

export type { BreadcrumbsProps } from './breadcrumbs'
export type { BreadcrumbsItemProps } from './breadcrumbs-item'
export type { BreadcrumbsSeparatorProps } from './breadcrumbs-separator'
export default Breadcrumbs as BreadcrumbsComponentType
