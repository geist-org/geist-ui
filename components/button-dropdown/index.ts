import ButtonDropdown from './button-dropdown'
import ButtonDropdownItem from './button-dropdown-item'

type ButtonDropdownType = typeof ButtonDropdown & {
  Item: typeof ButtonDropdownItem
}
;(ButtonDropdown as ButtonDropdownType).Item = ButtonDropdownItem

export type { ButtonDropdownProps, ButtonDropdownTypes } from './button-dropdown'
export type {
  ButtonDropdownItemProps,
  ButtonDropdownItemTypes,
} from './button-dropdown-item'
export default ButtonDropdown as ButtonDropdownType
