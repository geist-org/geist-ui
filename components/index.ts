// @ts-ignore
/// <reference types="styled-jsx" />

export { default as AutoComplete } from './auto-complete'
export type { AutoCompleteProps } from './auto-complete'

export { default as Avatar } from './avatar'
export type { AvatarProps, AvatarGroupProps } from './avatar'

export { default as Badge } from './badge'
export type { BadgeProps, BadgeAnchorProps } from './badge'

export { default as Breadcrumbs } from './breadcrumbs'
export type {
  BreadcrumbsProps,
  BreadcrumbsItemProps,
  BreadcrumbsSeparatorProps,
} from './breadcrumbs'

export { default as Button } from './button'
export type { ButtonProps } from './button'

export { default as ButtonDropdown } from './button-dropdown'
export type { ButtonDropdownProps, ButtonDropdownItemProps } from './button-dropdown'

export { default as ButtonGroup } from './button-group'
export type { ButtonGroupProps } from './button-group'

export { default as Capacity } from './capacity'
export type { CapacityProps } from './capacity'

export { default as Card } from './card'
export type { CardProps, CardContentProps, CardFooterProps } from './card'

export { default as Checkbox } from './checkbox'
export type { CheckboxProps, CheckboxGroupProps } from './checkbox'

export { default as Code } from './code'
export type { CodeProps } from './code'

export { default as Collapse } from './collapse'
export type { CollapseProps, CollapseGroupProps } from './collapse'

export { default as Description } from './description'
export type { DescriptionProps } from './description'

export { default as Display } from './display'
export type { DisplayProps } from './display'

export { default as Divider } from './divider'
export type { DividerProps } from './divider'

export { default as Dot } from './dot'
export type { DotProps } from './dot'

export { default as Drawer } from './drawer'
export type { DrawerProps } from './drawer'

export { default as Fieldset } from './fieldset'
export type {
  FieldsetProps,
  FieldsetTitleProps,
  FieldsetSubtitleProps,
  FieldsetGroupProps,
  FieldsetFooterProps,
  FieldsetContentProps,
} from './fieldset'

export { default as GeistProvider } from './geist-provider'
export type { GeistProviderProps } from './geist-provider'

export { default as Grid } from './grid'
export type { GridProps, GridContainerProps } from './grid'

export { default as Image } from './image'
export type { ImageProps, ImageBrowserProps } from './image'

export { default as Input } from './input'
export type { InputProps, InputPasswordProps } from './input'

export { default as Keyboard } from './keyboard'
export type { KeyboardProps } from './keyboard'

export { default as Link } from './link'
export type { LinkProps } from './link'

export { default as Loading } from './loading'
export type { LoadingProps } from './loading'

export { default as Modal } from './modal'
export type {
  ModalProps,
  ModalTitleProps,
  ModalSubtitleProps,
  ModalContentProps,
  ModalActionProps,
} from './modal'

export { default as Note } from './note'
export type { NoteProps } from './note'

export { default as Page } from './page'
export type {
  PageProps,
  PageHeaderProps,
  PageContentProps,
  PageFooterProps,
} from './page'

export { default as Pagination } from './pagination'
export type {
  PaginationProps,
  PaginationNextProps,
  PaginationPreviousProps,
} from './pagination'

export { default as Popover } from './popover'
export type { PopoverProps, PopoverItemProps } from './popover'

export { default as Progress } from './progress'
export type { ProgressProps } from './progress'

export { default as Radio } from './radio'
export type { RadioProps, RadioGroupProps, RadioDescriptionProps } from './radio'

export { default as Rating } from './rating'
export type { RatingProps } from './rating'

export { default as Select } from './select'
export type { SelectProps, SelectOptionProps } from './select'

export { default as Slider } from './slider'
export type { SliderProps } from './slider'

export { default as Snippet } from './snippet'
export type { SnippetProps } from './snippet'

export { default as Spacer } from './spacer'
export type { SpacerProps } from './spacer'

export { default as Spinner } from './spinner'
export type { SpinnerProps } from './spinner'

export { default as Table } from './table'
export type { TableProps, TableColumnProps } from './table'

export { default as Tabs } from './tabs'
export type { TabsProps } from './tabs'

export { default as Tag } from './tag'
export type { TagProps } from './tag'

export { default as Text } from './text'
export type { TextProps } from './text'

export { default as Textarea } from './textarea'
export type { TextareaProps } from './textarea'

export { default as Themes } from './themes'
export type { GeistUIThemes, GeistUserTheme } from './themes'

export { default as Toggle } from './toggle'
export type { ToggleProps } from './toggle'

export { default as Tooltip } from './tooltip'
export type { TooltipProps } from './tooltip'

export { default as Tree } from './tree'
export type { TreeProps } from './tree'

export { default as useAllThemes } from './use-all-themes'
export type { AllThemesConfig } from './use-all-themes'

export { default as useToasts } from './use-toasts'
export type { Toast, ToastInput, ToastAction, ToastLayout } from './use-toasts'

export { default as User } from './user'
export type { UserProps } from './user'

export { default as useBodyScroll } from './use-body-scroll'
export type { BodyScrollOptions } from './use-body-scroll'

export { default as useClipboard } from './use-clipboard'
export type { UseClipboardOptions } from './use-clipboard'

export { default as useMediaQuery } from './use-media-query'
export type { ResponsiveOptions, ResponsiveBreakpoint } from './use-media-query'

export { default as useKeyboard, KeyMod, KeyCode } from './use-keyboard'
export type { KeyboardOptions, UseKeyboardHandler } from './use-keyboard'

export { default as useInput } from './use-input'
export { default as useModal } from './use-modal'
export { default as useTabs } from './use-tabs'
export { default as useClickAway } from './use-click-away'
export { default as useCurrentState } from './use-current-state'
export { default as CssBaseline } from './css-baseline'
export { default as useTheme } from './use-theme'
export { default as useClasses } from './use-classes'
export { default as useScale } from './use-scale'
export { withScale, ScalePropKeys, ScaleContext } from './use-scale'
export type {
  ScaleProps,
  ScaleConfig,
  GetScalePropsFunction,
  GetAllScalePropsFunction,
} from './use-scale'

export { useRect } from './utils/layouts'
export type { ReactiveDomReact } from './utils/layouts'
export { default as Highlight } from './shared/highlight'
export type { HighlightProps } from './shared/highlight'
