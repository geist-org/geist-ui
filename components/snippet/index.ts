import Snippet from './snippet'

export type { SnippetProps, ToastTypes } from './snippet'
export type { CopyTypes, SnippetTypes } from '../utils/prop-types'
export default Snippet
