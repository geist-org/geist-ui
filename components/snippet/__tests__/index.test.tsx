import React from 'react'
import { mount } from 'enzyme'
import { Snippet } from 'components'

const command = 'yarn add @geist-ui/core'
const multiLine = ['cd project', 'now']

describe('Snippet', () => {
  beforeAll(() => {
    window.getSelection = jest.fn().mockImplementation(() => ({
      removeAllRanges: jest.fn(),
      addRange: jest.fn(),
    }))
    document.createRange = jest.fn().mockImplementation(() => ({
      selectNode: jest.fn(),
    }))
  })

  it('should render correctly', () => {
    const wrapper = mount(<Snippet text={command} />)
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should work with multi-line', () => {
    const wrapper = mount(<Snippet text={multiLine} />)
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should work with different styles', () => {
    const wrapper = mount(
      <div>
        <Snippet text={command} filled />
        <Snippet text={command} width="20%" />
        <Snippet text={command} type="secondary" />
        <Snippet text={command} type="success" filled />
      </div>,
    )
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('text should be copied', () => {
    document.execCommand = jest.fn()
    const wrapper = mount(<Snippet text={command} />)
    wrapper.find('.copy').simulate('click')
    expect(document.execCommand).toHaveBeenCalled()
    ;(document.execCommand as jest.Mock).mockRestore()
  })

  it('multi-line commands should be copied', () => {
    document.execCommand = jest.fn()
    const wrapper = mount(<Snippet text={multiLine} />)
    wrapper.find('.copy').simulate('click')
    expect(document.execCommand).toHaveBeenCalled()
    ;(document.execCommand as jest.Mock).mockRestore()
  })

  it('child commands should be copied', () => {
    document.execCommand = jest.fn()
    const wrapper = mount(<Snippet>{command}</Snippet>)
    wrapper.find('.copy').simulate('click')
    expect(document.execCommand).toHaveBeenCalled()
    ;(document.execCommand as jest.Mock).mockRestore()
  })

  it('should disable copy function', () => {
    const wrapper = mount(<Snippet text={command} copy="prevent" />)
    expect(wrapper.find('.copy').length).toBe(0)
  })

  it('should work with custom symbol', () => {
    const wrapper = mount(<Snippet text={command} symbol=">" />)
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()

    const emptySymbolWrapper = mount(<Snippet text={command} symbol=" " />)
    expect(emptySymbolWrapper.html()).toMatchSnapshot()

    const emptySymbolWrapper2 = mount(<Snippet text={command} symbol="" />)
    expect(emptySymbolWrapper2.html()).toEqual(emptySymbolWrapper.html())

    expect(() => emptySymbolWrapper.unmount()).not.toThrow()
    expect(() => emptySymbolWrapper2.unmount()).not.toThrow()
  })

  it('should work with custom toast', () => {
    document.execCommand = jest.fn()
    const wrapper = mount(
      <Snippet text={command} toastText="Code copied!" toastType="secondary" />,
    )
    wrapper.find('.copy').simulate('click')
    expect(document.execCommand).toHaveBeenCalled()
    ;(document.execCommand as jest.Mock).mockRestore()
  })

  afterAll(() => {
    ;(window.getSelection as jest.Mock).mockRestore()
    ;(document.createRange as jest.Mock).mockRestore()
  })
})
