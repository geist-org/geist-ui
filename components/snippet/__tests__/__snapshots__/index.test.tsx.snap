// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snippet should render correctly 1`] = `
"<div class=\\"snippet\\"><pre>yarn add @geist-ui/core</pre><div class=\\"copy\\"><svg viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" fill=\\"none\\" shape-rendering=\\"geometricPrecision\\" style=\\"color: currentcolor;\\"><path d=\\"M8 17.929H6c-1.105 0-2-.912-2-2.036V5.036C4 3.91 4.895 3 6 3h8c1.105 0 2 .911 2 2.036v1.866m-6 .17h8c1.105 0 2 .91 2 2.035v10.857C20 21.09 19.105 22 18 22h-8c-1.105 0-2-.911-2-2.036V9.107c0-1.124.895-2.036 2-2.036z\\"></path><style>
        svg {
          width: calc(var(--snippet-font-size) * 1.69);
          height: calc(var(--snippet-font-size) * 1.69);
        }
      </style></svg></div><style>
        .snippet {
          position: relative;
          max-width: 100%;
          color: #000;
          background-color: #fff;
          border: 1px solid #eaeaea;
          border-radius: 6px;
          --snippet-font-size: calc(0.8125 * 16px);
          --snippet-padding-top: calc(0.667 * 16px);
          font-size: var(--snippet-font-size);
          width: initial;
          height: auto;
          padding: calc(0.667 * 16px) calc(2.667 * 16px) calc(0.667 * 16px)
            calc(0.667 * 16px);
          margin: 0 0 0 0;
        }

        pre {
          margin: 0;
          padding: 0;
          border: none;
          background-color: transparent;
          color: #000;
          font-size: inherit;
        }

        pre::before {
          content: '$ ';
          user-select: none;
        }

        pre :global(*) {
          margin: 0;
          padding: 0;
          font-size: inherit;
          color: inherit;
        }

        .copy {
          position: absolute;
          right: 0;
          top: 0;
          bottom: 0;
          height: calc(100% - 2px);
          background-color: #fff;
          display: inline-flex;
          justify-content: center;
          align-items: center;
          width: calc(3.281 * var(--snippet-font-size));
          color: inherit;
          transition: opacity 150ms ease 0s;
          border-radius: 6px;
          cursor: pointer;
          user-select: none;
          padding-top: 0;
          opacity: 0.65;
        }

        .copy:hover {
          opacity: 1;
        }
      </style></div>"
`;

exports[`Snippet should work with custom symbol 1`] = `
"<div class=\\"snippet\\"><pre>yarn add @geist-ui/core</pre><div class=\\"copy\\"><svg viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" fill=\\"none\\" shape-rendering=\\"geometricPrecision\\" style=\\"color: currentcolor;\\"><path d=\\"M8 17.929H6c-1.105 0-2-.912-2-2.036V5.036C4 3.91 4.895 3 6 3h8c1.105 0 2 .911 2 2.036v1.866m-6 .17h8c1.105 0 2 .91 2 2.035v10.857C20 21.09 19.105 22 18 22h-8c-1.105 0-2-.911-2-2.036V9.107c0-1.124.895-2.036 2-2.036z\\"></path><style>
        svg {
          width: calc(var(--snippet-font-size) * 1.69);
          height: calc(var(--snippet-font-size) * 1.69);
        }
      </style></svg></div><style>
        .snippet {
          position: relative;
          max-width: 100%;
          color: #000;
          background-color: #fff;
          border: 1px solid #eaeaea;
          border-radius: 6px;
          --snippet-font-size: calc(0.8125 * 16px);
          --snippet-padding-top: calc(0.667 * 16px);
          font-size: var(--snippet-font-size);
          width: initial;
          height: auto;
          padding: calc(0.667 * 16px) calc(2.667 * 16px) calc(0.667 * 16px)
            calc(0.667 * 16px);
          margin: 0 0 0 0;
        }

        pre {
          margin: 0;
          padding: 0;
          border: none;
          background-color: transparent;
          color: #000;
          font-size: inherit;
        }

        pre::before {
          content: '> ';
          user-select: none;
        }

        pre :global(*) {
          margin: 0;
          padding: 0;
          font-size: inherit;
          color: inherit;
        }

        .copy {
          position: absolute;
          right: 0;
          top: 0;
          bottom: 0;
          height: calc(100% - 2px);
          background-color: #fff;
          display: inline-flex;
          justify-content: center;
          align-items: center;
          width: calc(3.281 * var(--snippet-font-size));
          color: inherit;
          transition: opacity 150ms ease 0s;
          border-radius: 6px;
          cursor: pointer;
          user-select: none;
          padding-top: 0;
          opacity: 0.65;
        }

        .copy:hover {
          opacity: 1;
        }
      </style></div>"
`;

exports[`Snippet should work with custom symbol 2`] = `
"<div class=\\"snippet\\"><pre>yarn add @geist-ui/core</pre><div class=\\"copy\\"><svg viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" fill=\\"none\\" shape-rendering=\\"geometricPrecision\\" style=\\"color: currentcolor;\\"><path d=\\"M8 17.929H6c-1.105 0-2-.912-2-2.036V5.036C4 3.91 4.895 3 6 3h8c1.105 0 2 .911 2 2.036v1.866m-6 .17h8c1.105 0 2 .91 2 2.035v10.857C20 21.09 19.105 22 18 22h-8c-1.105 0-2-.911-2-2.036V9.107c0-1.124.895-2.036 2-2.036z\\"></path><style>
        svg {
          width: calc(var(--snippet-font-size) * 1.69);
          height: calc(var(--snippet-font-size) * 1.69);
        }
      </style></svg></div><style>
        .snippet {
          position: relative;
          max-width: 100%;
          color: #000;
          background-color: #fff;
          border: 1px solid #eaeaea;
          border-radius: 6px;
          --snippet-font-size: calc(0.8125 * 16px);
          --snippet-padding-top: calc(0.667 * 16px);
          font-size: var(--snippet-font-size);
          width: initial;
          height: auto;
          padding: calc(0.667 * 16px) calc(2.667 * 16px) calc(0.667 * 16px)
            calc(0.667 * 16px);
          margin: 0 0 0 0;
        }

        pre {
          margin: 0;
          padding: 0;
          border: none;
          background-color: transparent;
          color: #000;
          font-size: inherit;
        }

        pre::before {
          content: '';
          user-select: none;
        }

        pre :global(*) {
          margin: 0;
          padding: 0;
          font-size: inherit;
          color: inherit;
        }

        .copy {
          position: absolute;
          right: 0;
          top: 0;
          bottom: 0;
          height: calc(100% - 2px);
          background-color: #fff;
          display: inline-flex;
          justify-content: center;
          align-items: center;
          width: calc(3.281 * var(--snippet-font-size));
          color: inherit;
          transition: opacity 150ms ease 0s;
          border-radius: 6px;
          cursor: pointer;
          user-select: none;
          padding-top: 0;
          opacity: 0.65;
        }

        .copy:hover {
          opacity: 1;
        }
      </style></div>"
`;

exports[`Snippet should work with different styles 1`] = `
"<div><div class=\\"snippet\\"><pre>yarn add @geist-ui/core</pre><div class=\\"copy\\"><svg viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" fill=\\"none\\" shape-rendering=\\"geometricPrecision\\" style=\\"color: currentcolor;\\"><path d=\\"M8 17.929H6c-1.105 0-2-.912-2-2.036V5.036C4 3.91 4.895 3 6 3h8c1.105 0 2 .911 2 2.036v1.866m-6 .17h8c1.105 0 2 .91 2 2.035v10.857C20 21.09 19.105 22 18 22h-8c-1.105 0-2-.911-2-2.036V9.107c0-1.124.895-2.036 2-2.036z\\"></path><style>
        svg {
          width: calc(var(--snippet-font-size) * 1.69);
          height: calc(var(--snippet-font-size) * 1.69);
        }
      </style></svg></div><style>
        .snippet {
          position: relative;
          max-width: 100%;
          color: #000;
          background-color: #fff;
          border: 1px solid #eaeaea;
          border-radius: 6px;
          --snippet-font-size: calc(0.8125 * 16px);
          --snippet-padding-top: calc(0.667 * 16px);
          font-size: var(--snippet-font-size);
          width: initial;
          height: auto;
          padding: calc(0.667 * 16px) calc(2.667 * 16px) calc(0.667 * 16px)
            calc(0.667 * 16px);
          margin: 0 0 0 0;
        }

        pre {
          margin: 0;
          padding: 0;
          border: none;
          background-color: transparent;
          color: #000;
          font-size: inherit;
        }

        pre::before {
          content: '$ ';
          user-select: none;
        }

        pre :global(*) {
          margin: 0;
          padding: 0;
          font-size: inherit;
          color: inherit;
        }

        .copy {
          position: absolute;
          right: 0;
          top: 0;
          bottom: 0;
          height: calc(100% - 2px);
          background-color: #fff;
          display: inline-flex;
          justify-content: center;
          align-items: center;
          width: calc(3.281 * var(--snippet-font-size));
          color: inherit;
          transition: opacity 150ms ease 0s;
          border-radius: 6px;
          cursor: pointer;
          user-select: none;
          padding-top: 0;
          opacity: 0.65;
        }

        .copy:hover {
          opacity: 1;
        }
      </style></div><div class=\\"snippet\\"><pre>yarn add @geist-ui/core</pre><div class=\\"copy\\"><svg viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" fill=\\"none\\" shape-rendering=\\"geometricPrecision\\" style=\\"color: currentcolor;\\"><path d=\\"M8 17.929H6c-1.105 0-2-.912-2-2.036V5.036C4 3.91 4.895 3 6 3h8c1.105 0 2 .911 2 2.036v1.866m-6 .17h8c1.105 0 2 .91 2 2.035v10.857C20 21.09 19.105 22 18 22h-8c-1.105 0-2-.911-2-2.036V9.107c0-1.124.895-2.036 2-2.036z\\"></path><style>
        svg {
          width: calc(var(--snippet-font-size) * 1.69);
          height: calc(var(--snippet-font-size) * 1.69);
        }
      </style></svg></div><style>
        .snippet {
          position: relative;
          max-width: 100%;
          color: #000;
          background-color: #fff;
          border: 1px solid #eaeaea;
          border-radius: 6px;
          --snippet-font-size: calc(0.8125 * 16px);
          --snippet-padding-top: calc(0.667 * 16px);
          font-size: var(--snippet-font-size);
          width: 20%;
          height: auto;
          padding: calc(0.667 * 16px) calc(2.667 * 16px) calc(0.667 * 16px)
            calc(0.667 * 16px);
          margin: 0 0 0 0;
        }

        pre {
          margin: 0;
          padding: 0;
          border: none;
          background-color: transparent;
          color: #000;
          font-size: inherit;
        }

        pre::before {
          content: '$ ';
          user-select: none;
        }

        pre :global(*) {
          margin: 0;
          padding: 0;
          font-size: inherit;
          color: inherit;
        }

        .copy {
          position: absolute;
          right: 0;
          top: 0;
          bottom: 0;
          height: calc(100% - 2px);
          background-color: #fff;
          display: inline-flex;
          justify-content: center;
          align-items: center;
          width: calc(3.281 * var(--snippet-font-size));
          color: inherit;
          transition: opacity 150ms ease 0s;
          border-radius: 6px;
          cursor: pointer;
          user-select: none;
          padding-top: 0;
          opacity: 0.65;
        }

        .copy:hover {
          opacity: 1;
        }
      </style></div><div class=\\"snippet\\"><pre>yarn add @geist-ui/core</pre><div class=\\"copy\\"><svg viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" fill=\\"none\\" shape-rendering=\\"geometricPrecision\\" style=\\"color: currentcolor;\\"><path d=\\"M8 17.929H6c-1.105 0-2-.912-2-2.036V5.036C4 3.91 4.895 3 6 3h8c1.105 0 2 .911 2 2.036v1.866m-6 .17h8c1.105 0 2 .91 2 2.035v10.857C20 21.09 19.105 22 18 22h-8c-1.105 0-2-.911-2-2.036V9.107c0-1.124.895-2.036 2-2.036z\\"></path><style>
        svg {
          width: calc(var(--snippet-font-size) * 1.69);
          height: calc(var(--snippet-font-size) * 1.69);
        }
      </style></svg></div><style>
        .snippet {
          position: relative;
          max-width: 100%;
          color: #666;
          background-color: #fff;
          border: 1px solid #666;
          border-radius: 6px;
          --snippet-font-size: calc(0.8125 * 16px);
          --snippet-padding-top: calc(0.667 * 16px);
          font-size: var(--snippet-font-size);
          width: initial;
          height: auto;
          padding: calc(0.667 * 16px) calc(2.667 * 16px) calc(0.667 * 16px)
            calc(0.667 * 16px);
          margin: 0 0 0 0;
        }

        pre {
          margin: 0;
          padding: 0;
          border: none;
          background-color: transparent;
          color: #666;
          font-size: inherit;
        }

        pre::before {
          content: '$ ';
          user-select: none;
        }

        pre :global(*) {
          margin: 0;
          padding: 0;
          font-size: inherit;
          color: inherit;
        }

        .copy {
          position: absolute;
          right: 0;
          top: 0;
          bottom: 0;
          height: calc(100% - 2px);
          background-color: #fff;
          display: inline-flex;
          justify-content: center;
          align-items: center;
          width: calc(3.281 * var(--snippet-font-size));
          color: inherit;
          transition: opacity 150ms ease 0s;
          border-radius: 6px;
          cursor: pointer;
          user-select: none;
          padding-top: 0;
          opacity: 0.65;
        }

        .copy:hover {
          opacity: 1;
        }
      </style></div><div class=\\"snippet\\"><pre>yarn add @geist-ui/core</pre><div class=\\"copy\\"><svg viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" fill=\\"none\\" shape-rendering=\\"geometricPrecision\\" style=\\"color: currentcolor;\\"><path d=\\"M8 17.929H6c-1.105 0-2-.912-2-2.036V5.036C4 3.91 4.895 3 6 3h8c1.105 0 2 .911 2 2.036v1.866m-6 .17h8c1.105 0 2 .91 2 2.035v10.857C20 21.09 19.105 22 18 22h-8c-1.105 0-2-.911-2-2.036V9.107c0-1.124.895-2.036 2-2.036z\\"></path><style>
        svg {
          width: calc(var(--snippet-font-size) * 1.69);
          height: calc(var(--snippet-font-size) * 1.69);
        }
      </style></svg></div><style>
        .snippet {
          position: relative;
          max-width: 100%;
          color: #fff;
          background-color: #0070f3;
          border: 1px solid #0070f3;
          border-radius: 6px;
          --snippet-font-size: calc(0.8125 * 16px);
          --snippet-padding-top: calc(0.667 * 16px);
          font-size: var(--snippet-font-size);
          width: initial;
          height: auto;
          padding: calc(0.667 * 16px) calc(2.667 * 16px) calc(0.667 * 16px)
            calc(0.667 * 16px);
          margin: 0 0 0 0;
        }

        pre {
          margin: 0;
          padding: 0;
          border: none;
          background-color: transparent;
          color: #fff;
          font-size: inherit;
        }

        pre::before {
          content: '$ ';
          user-select: none;
        }

        pre :global(*) {
          margin: 0;
          padding: 0;
          font-size: inherit;
          color: inherit;
        }

        .copy {
          position: absolute;
          right: 0;
          top: 0;
          bottom: 0;
          height: calc(100% - 2px);
          background-color: #0070f3;
          display: inline-flex;
          justify-content: center;
          align-items: center;
          width: calc(3.281 * var(--snippet-font-size));
          color: inherit;
          transition: opacity 150ms ease 0s;
          border-radius: 6px;
          cursor: pointer;
          user-select: none;
          padding-top: 0;
          opacity: 0.65;
        }

        .copy:hover {
          opacity: 1;
        }
      </style></div></div>"
`;

exports[`Snippet should work with multi-line 1`] = `
"<div class=\\"snippet\\"><pre>cd project</pre><pre>now</pre><div class=\\"copy\\"><svg viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" fill=\\"none\\" shape-rendering=\\"geometricPrecision\\" style=\\"color: currentcolor;\\"><path d=\\"M8 17.929H6c-1.105 0-2-.912-2-2.036V5.036C4 3.91 4.895 3 6 3h8c1.105 0 2 .911 2 2.036v1.866m-6 .17h8c1.105 0 2 .91 2 2.035v10.857C20 21.09 19.105 22 18 22h-8c-1.105 0-2-.911-2-2.036V9.107c0-1.124.895-2.036 2-2.036z\\"></path><style>
        svg {
          width: calc(var(--snippet-font-size) * 1.69);
          height: calc(var(--snippet-font-size) * 1.69);
        }
      </style></svg></div><style>
        .snippet {
          position: relative;
          max-width: 100%;
          color: #000;
          background-color: #fff;
          border: 1px solid #eaeaea;
          border-radius: 6px;
          --snippet-font-size: calc(0.8125 * 16px);
          --snippet-padding-top: calc(0.667 * 16px);
          font-size: var(--snippet-font-size);
          width: initial;
          height: auto;
          padding: calc(0.667 * 16px) calc(2.667 * 16px) calc(0.667 * 16px)
            calc(0.667 * 16px);
          margin: 0 0 0 0;
        }

        pre {
          margin: 0;
          padding: 0;
          border: none;
          background-color: transparent;
          color: #000;
          font-size: inherit;
        }

        pre::before {
          content: '$ ';
          user-select: none;
        }

        pre :global(*) {
          margin: 0;
          padding: 0;
          font-size: inherit;
          color: inherit;
        }

        .copy {
          position: absolute;
          right: 0;
          top: 0;
          bottom: 0;
          height: calc(100% - 2px);
          background-color: #fff;
          display: inline-flex;
          justify-content: center;
          align-items: flex-start;
          width: calc(3.281 * var(--snippet-font-size));
          color: inherit;
          transition: opacity 150ms ease 0s;
          border-radius: 6px;
          cursor: pointer;
          user-select: none;
          padding-top: var(--snippet-padding-top);
          opacity: 0.65;
        }

        .copy:hover {
          opacity: 1;
        }
      </style></div>"
`;
